flocker/control/_protocol.py

```diff
@@ -727,7 +859,11 @@ def disconnected(self, connection):
 
         :param ControlAMP connection: The lost connection.
         """
-        self.connections.remove(connection)
+        self._connections.remove(connection)
+        if connection in self._connections_pending_update:
+            self._connections_pending_update.remove(connection)
```

Comment:
:question: This looks like it might be something we missed in https://github.com/ClusterHQ/flocker/pull/2782 rather than directly related to this branch.